x-pack/platform/plugins/shared/fields_metadata/common/fields_metadata/types.ts

```diff
@@ -33,6 +33,15 @@ export const multiFieldRT = rt.type({
   type: rt.string,
 });
 
+export const otelPropertyRT = rt.partial({
+  relation: rt.string,
+  stability: rt.string,
+  attribute: rt.string,
+  otlp_field: rt.string,
+  metric: rt.string,
+  note: rt.string,
+});
+
```

Comment:
> [!IMPORTANT]
> I left it loosely typed as most of the other properties to avoid validation errors when property values scales with new possible values and avoid breaking the API

Reply:
I think it would be good to make these a bit more specific since it's gonna be difficult to consume otherwise. 

It shouldn't be an issue if values change in the `@elastic/ecs` package since we control the version we pull in. 

We could also add a unit test to verify that the type definitions match each field so that the CI breaks when someone updates the package without updating the type definitions. 

Doesn't need to block this PR though, we can carry out that work as part of https://github.com/elastic/kibana/issues/224103

Reply:
Amazing, looks great!